2_observations/src/munge_diversion_data.R

```diff
@@ -0,0 +1,239 @@
+           reservoir = 'Cannonsville') %>%
+    filter(!is.na(date)) %>%
+    filter(!is.na(diversion_mgd))
+
+  # write data to csv
+  daily_out <- bind_rows(cleaned_res_dat, pepacton_dat, cannonsville_dat) %>%
+    select(-dayofmonth, -year, -month, -doy) %>%
+    rename(source_file = filename)
+
+  write_csv(x = daily_out, file = as_data_file(out_ind))
+  gd_put(out_ind)
+}
+
+read_munge_monthlies <- function(in_file, out_ind) {
+  # function to read in and munge monthly files that were
+  # either manually entered by the NY WSC
+  # or by Sam
+
+  # import manually-entered (by us) historical data
+  manual_monthly <- read.csv(in_file) %>%
+    rename(diversion_cfs = diversion) %>%
+    mutate(filename = '2_observations/in/reservoir data/manually_extracted_diversions.csv')
+
+  # import pre-entered (by NY WSC) monthly historical data
+  my_colnames <- c('month', 'daysinmonth', 'res_level_ft',
+                   'stor_above_sill_mg', 'gross_storage_mg',
+                   'storage_change_mgd', 'storage_change_cfs',
+                   'diversion_mgd', 'diversion_cfs')
+
+  root_dir <- '2_observations/in/reservoir data/'
+  all_files <- list.files(root_dir)
+
+  monthly_files <- all_files[grepl('.xls', all_files)]
+
+  # map over all monthly files, keep track of filename for parsing
+  dat <- monthly_files %>%
+    set_names() %>%
+    purrr::map_df(~readxl::read_excel(file.path(root_dir, .x),
+                                      range = 'A12:I24',
+                                      col_names = my_colnames), .id = 'filename')
+
+  # pull out identifying info from filename
+  dat_cleaned <- dat %>%
+    filter(!month %in% 'Calendar Year') %>%
+    mutate(wy = as.numeric(gsub('(.*WY)(\\d{4})(.xls)', '\\2', filename)),
+           year = ifelse(month %in% c( 'October', 'November', 'December'), wy - 1, wy),
+           reservoir = ifelse(grepl('01424997', filename), 'Cannonsville', 'Pepacton')) %>%
+    filter(!year %in% 2021)
+
+  # bind monthly data
+  out_dat <- bind_rows(dat_cleaned, manual_monthly) %>%
+    select(-daysinmonth, -wy)
+
+  write_csv(out_dat, as_data_file(out_ind))
+  gd_put(out_ind)
+}
+
+interpolate_to_daily <- function(out_ind, daily_ind, monthly_ind, mgd_to_cms, cfs_to_cms, mg_to_cm, ft_to_m) {
+  daily <- read_csv(sc_retrieve(daily_ind))
+
+  monthly <- read_csv(sc_retrieve(monthly_ind)) %>%
+    rename(source_file = filename) %>%
+    mutate(diversion_cms = diversion_cfs*cfs_to_cms) %>%
+    select(-storage_change_cfs, -diversion_cfs, -diversion_mgd)
+
+  # create a dataframe with all dates in timeseries for each reservoir
+  all_dates <- data.frame(date = seq(from = as.Date('1979-10-01'),
+                                          to = as.Date('2020-12-31'), by = 1)) %>%
+    mutate(year = lubridate::year(date),
+           month = lubridate::month(date, label = TRUE, abbr = FALSE))
+
+  all_dates <- bind_rows(mutate(all_dates, reservoir = 'Pepacton'), mutate(all_dates, reservoir = 'Cannonsville'))
```

Comment:
I recently learned about a cool feature of `bind_rows`, the `.id` argument, that could simplify this syntax if desired:
```r
> bind_rows(Pepacton=tibble(x=1:3), Cannonsville=tibble(x=4:6), .id='reservoir')
# A tibble: 6 x 2
  reservoir        x
  <chr>        <int>
1 Pepacton         1
2 Pepacton         2
3 Pepacton         3
4 Cannonsville     4
5 Cannonsville     5
6 Cannonsville     6
```

Reply:
Oh cool! I just learned the `.id` in `map_df` and used in this PR, very cool you can use it in `bind_rows` (and am betting `map_df` is just passing `.id` to `bind_rows`).